addons/flannel/template/base/install.sh

```diff
@@ -307,20 +307,72 @@ function flannel_pods_remain() {
 }
 
 function remove_weave() {
-    # firstnode only
-    kubectl -n kube-system delete daemonset weave-net
-    kubectl -n kube-system delete rolebinding weave-net
-    kubectl -n kube-system delete role weave-net
-    kubectl delete clusterrolebinding weave-net
-    kubectl delete clusterrole weave-net
-    kubectl -n kube-system delete serviceaccount weave-net
-    kubectl -n kube-system delete secret weave-passwd
+    local resources=("daemonset.apps/weave-net"
+                     "rolebinding.rbac.authorization.k8s.io/weave-net"
+                     "role.rbac.authorization.k8s.io/weave-net"
+                     "clusterrolebinding.rbac.authorization.k8s.io/weave-net"
+                     "clusterrole.rbac.authorization.k8s.io/weave-net"
+                     "serviceaccount/weave-net"
+                     "secret/weave-passwd")
+
+    for resource in "${resources[@]}"; do
+        if ! delete_resource_gracefully "kube-system" "$resource"; then
+            logWarn "$resource is still present after the deletion attempt with a grace period of 1 minute."
+        fi
+    done
 
     # all nodes
     ip link delete weave
+
     rm -rf /var/lib/weave
```

Comment:
see https://github.com/replicatedhq/kURL/pull/4555/files#diff-dd06f6bd0012cb6a930396f707affe1ce3a7ab8edd02145c46c2f334433fa11dR325 same thing here (idempotently run `ip link delete weave` only if the device exists)

Reply:
@CpuID Lets to do it in another PR so that we make eiaser the review and we ensure that we are updating accordingly all places by once. https://github.com/replicatedhq/kURL/pull/4576